craft_archives/repo/package_repository.py

```diff
@@ -222,71 +224,81 @@ class PackageRepositoryApt(PackageRepository):
     """An APT package repository."""
 
     url: Union[AnyUrl, FileUrl]
-    key_id: KeyIdStr = pydantic.Field(alias="key-id")
-    architectures: Optional[List[str]]
-    formats: Optional[List[Literal["deb", "deb-src"]]]
-    path: Optional[str]
-    components: Optional[List[str]]
-    key_server: Optional[str] = pydantic.Field(alias="key-server")
-    suites: Optional[List[str]]
-
-    # Customize some of the validation error messages
-    class Config(PackageRepository.Config):  # noqa: D106 - no docstring needed
-        error_msg_templates = {
-            "value_error.any_str.min_length": "Invalid URL; URLs must be non-empty strings"
-        }
+    key_id: Annotated[
+        str, StringConstraints(min_length=40, max_length=40, pattern=r"^[0-9A-F]{40}$")
+    ] = Field(alias="key-id")
+    architectures: Optional[List[str]] = None
+    formats: Optional[List[Literal["deb", "deb-src"]]] = None
+    path: Optional[str] = None
+    components: Optional[List[str]] = None
+    key_server: Optional[Annotated[str, Field(alias="key-server")]] = None
+    suites: Optional[List[str]] = None
 
     @property
     def name(self) -> str:
         """Get the repository name."""
-        return re.sub(r"\W+", "_", self.url)
+        return re.sub(r"\W+", "_", str(self.url).rstrip("/"))
```

Comment:
with the amount of `str(self.url).rstrip("/")` that this code does maybe we should just add a property `stripped_url`

Reply:
I just removed all those extra occurences of rstrip.